src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java

```diff
@@ -311,6 +316,37 @@ public Response readFile(@PathParam("uuid") final UUID uuid, @PathParam("versie"
         }
     }
 
+    @GET
+    @Path("/informatieobject/download/zip")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public Response readFilesAsZip(@QueryParam("uuids") final String uuids) {
+        final List<String> enkelvoudigInformatieObjectUUIDS = Arrays.stream(uuids.split(",")).toList();
+
+        StreamingOutput streamingOutput = outputStream -> {
+            ZipOutputStream zipOutputStream = new ZipOutputStream(new BufferedOutputStream(outputStream));
+            enkelvoudigInformatieObjectUUIDS.forEach(stringUuid -> {
+                final UUID uuid = UUID.fromString(stringUuid);
+                final EnkelvoudigInformatieobject enkelvoudigInformatieobject = drcClientService.readEnkelvoudigInformatieobject(uuid);
+                final ByteArrayInputStream inhoud = drcClientService.downloadEnkelvoudigInformatieobject(uuid);
+                final ZipEntry zipEntry = new ZipEntry(enkelvoudigInformatieobject.getBestandsnaam());
+                try {
+                    zipOutputStream.putNextEntry(zipEntry);
+                    zipOutputStream.write(inhoud.readAllBytes());
+                    zipOutputStream.closeEntry();
+                } catch (final IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+            zipOutputStream.close();
+            outputStream.flush();
+            outputStream.close();
+        };
+
```

Comment:
Ik zou hier met het oog op de resource leaks wel even de try met resources gebruiken voor alle closeables. En die moeten dus in de try elke een variabele krijgen (je mag ze niet nesten).

Reply:
Aangepast